src/main/java/de/bonndan/nivio/input/rancher1/APIWalker.java

```diff
@@ -0,0 +1,135 @@
+        Rancher.Config config = getConfig(reference);
+        this.rancher = new Rancher(config);
+        this.reference = reference;
+    }
+
+    public List<ItemDescription> getDescriptions() {
+
+        String projectName = (String) reference.getProperty("projectName");
+        Project project = getProject(projectName)
+                .orElseThrow(() -> new ProcessingException(reference.getLandscapeDescription(), "Project " + projectName + "not found"));
+        String accountId = project.getId();
+
+
+        Map<String, Stack> stacks = getStacksById(accountId);
+        List<Service> services = getServices(accountId);
+
+        return asDescriptions(services, stacks);
+    }
+
+    private List<Service> getServices(final String accountId) {
+        ServiceService service = rancher.type(ServiceService.class);
+        try {
+            return service.list().execute().body().getData().stream()
+                    .filter(service1 -> service1.getAccountId().equals(accountId))
+                    .collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new ProcessingException(reference.getLandscapeDescription(), "Could not load services from Rancher API", e);
+        }
+    }
+
+    private Map<String, Stack> getStacksById(String accountId) {
+        StackService stackService = rancher.type(StackService.class);
+        Map<String, Stack> stacks = new HashMap<>();
+        try {
+            stackService.list().execute().body().getData().stream()
+                    .filter(stack -> stack.getAccountId().equals(accountId))
+                    .forEach(stack -> stacks.put(stack.getId(), stack));
+            return stacks;
+        } catch (IOException e) {
+            throw new ProcessingException(reference.getLandscapeDescription(), "Could not access Rancher API", e);
+        }
+    }
+
+    private Optional<Project> getProject(String projectName) {
+        ProjectService projectService = rancher.type(ProjectService.class);
+        List<Project> projects = null;
+        try {
+            projects = projectService.list().execute().body().getData();
+        } catch (IOException e) {
+            throw new ProcessingException("Could not load projects", e);
+        }
+        return projects.stream()
+                .filter(project -> StringUtils.isEmpty(projectName) || project.getName().equals(projectName))
+                .findFirst();
+    }
+
+    private List<ItemDescription> asDescriptions(List<Service> data, final Map<String, Stack> stacks) {
+        List<ItemDescription> descriptions = new ArrayList<>();
+
+        data.forEach(service -> {
+            ItemDescription item = new ItemDescription();
+            item.setIdentifier(service.getId());
+            item.setName(service.getName());
+            item.setScale(String.valueOf(service.getScale()));
+            Stack stack = stacks.get(service.getStackId());
+            if (stack != null)
```

Comment:
Mal rein fachlich gefragt: Kann es in Rancher überhaupt vorkommen, dass ein Service nicht in irgendeinem Stack ist? Wenn dieser Fall auftrifft ist glaube ich was größeres im Argen.